### PR TITLE
Initiate aws modules

### PR DESCRIPTION
**Pull request by phillebaba**

### CODE REVIEW

File: modules/aws/core/main.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  required_version = "0.13.5"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "3.20.0"
+    }
+  }
+}
+
+data "aws_availability_zones" "available" {
```
Comment (simongottschlag) [inline modules/aws/core/main.tf:11]: `this` or `az`

File: modules/aws/core/vpc.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  vpc = true
+
+  tags = {
+    Name        = "${var.name}-${each.value}"
+    Environment = var.environment
+  }
+}
+
+resource "aws_nat_gateway" "nat" {
```
Comment (simongottschlag) [inline modules/aws/core/vpc.tf:70]: `this` ?

File: modules/aws/eks/external-secrets.tf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#      "rbac.authorization.k8s.io/aggregate-to-edit" = "true"
+#    }
+#  }
+#
+#  rule {
+#    api_groups = ["kubernetes-client.io"]
+#    resources  = ["externalsecrets"]
+#    verbs      = ["get", "list", "watch", "create", "update", "patch", "delete"]
+#  }
+#}
```
Comment (simongottschlag) [inline modules/aws/eks/external-secrets.tf:68]: Can this be removed?

File: modules/aws/eks/iam.tf [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+#
+#resource "null_resource" "aws_auth" {
+#  triggers = {
+#    aws_auth = base64encode(data.template_file.aws_auth.rendered)
+#  }
+#
+#  provisioner "local-exec" {
+#    command = "echo '${data.template_file.aws_auth.rendered}' | kubectl apply --insecure-skip-tls-verify=true --server='${aws_eks_cluster.eks.endpoint}' --token='${data.aws_eks_cluster_auth.eks.token}' -f -"
+#  }
+#}
```
Comment (simongottschlag) [inline modules/aws/eks/iam.tf:181]: Can this be removed?

File: modules/aws/eks/main.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      version = "3.0.0"
+    }
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "available" {
```
Comment (simongottschlag) [inline modules/aws/eks/main.tf:19]: `this` or `az`

File: modules/aws/eks/variables.tf [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+#variable "namespaces" {
+#  description = "The namespaces that should be created in Kubernetes."
+#  type = list(
+#    object({
+#      name                    = string
+#      delegate_resource_group = bool
+#    })
+#  )
+#}
```
Comment (simongottschlag) [inline modules/aws/eks/variables.tf:46]: Can this be removed?

File: modules/kubernetes/eks-core/main.tf [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  }
+}
+
+data "aws_region" "current" {}
+
+#locals {
+#  # Namespace to create service accounts in
+#  service_accounts_namespace = "service-accounts"
+#}
```
Comment (simongottschlag) [inline modules/kubernetes/eks-core/main.tf:43]: Can this be removed?

File: modules/kubernetes/eks-core/modules.tf [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    for s in ["opa-gatekeeper"] :
+    s => s
+    if var.opa_gatekeeper_enabled
+  }
+
+  source = "../../kubernetes/opa-gatekeeper"
+
+  exclude = [
+    {
+      excluded_namespaces = ["kube-system", "gatekeeper-system", "kyverno", "cert-manager", "ingress-nginx", "velero", "flux-system", "external-dns", "external-secrets"]
```
Comment (simongottschlag) [inline modules/kubernetes/eks-core/modules.tf:13]: kyverno?

File: modules/kubernetes/eks-core/variables.tf [hunk 9]
```diff
@@ -0,0 +1,128 @@
+variable "environment" {
+  description = "The environment name to use for the deploy"
+  type        = string
+}
+
+#variable "name" {
+#  description = "The name to use for the deploy"
+#  type        = string
+#}
```
Comment (simongottschlag) [inline modules/kubernetes/eks-core/variables.tf:9]: can this be removed?

File: modules/aws/eks/external-secrets.tf (hunk 3 above)
Comment (phillebaba) [reply modules/aws/eks/external-secrets.tf:68]: it will be in another pr

File: modules/aws/core/vpc.tf (hunk 2 above)
Comment (phillebaba) [reply modules/aws/core/vpc.tf:70]: no because it has an eip linked to it

File: modules/aws/eks/iam.tf (hunk 4 above)
Comment (phillebaba) [reply modules/aws/eks/iam.tf:181]: it will be moved in another pr

**Review by simongottschlag — APPROVED** [review]
LGTM

**Pull request merged**